src/btc.rs

```diff
@@ -142,13 +164,27 @@ impl UnsignedTx {
         secp: &Secp256k1<C>,
         signature: &Signature,
         merkle_root: Option<TapNodeHash>,
+    ) -> Result<(), Error> {
+        let witness = Witness::p2tr_key_spend(signature);
+        self.verify_witness(secp, witness, merkle_root)
+    }
+
+    /// Tests if the provided taproot [`Signature`] is valid for spending the
+    /// signers' UTXO. This function will return  [`Error::BitcoinConsensus`]
+    /// error if the signature fails verification, passing the underlying error
+    /// from [`bitcoinconsensus`].
```

Comment:
This description, while kinda true, seems inaccurate. The function is provided a signature only in the witness data. Perhaps it should mention that it checks that the provided witness data would satisfy bitcoin's consensus rules for the input in this transaction.

Reply:
I updated the comment block here and in the above `fn`